packages/driver/src/cy/ensures.ts

```diff
@@ -331,6 +331,15 @@ export default {
       const cmd = state('current').get('name')
 
       if (!$dom.isDescendent($el1, $el2)) {
+        // https://github.com/cypress-io/cypress/issues/18008
+        // when an element inside a shadow root is covered by its shadow host
+        if (
+          $dom.isWithinShadowRoot($el1.get(0)) &&
+          $el1.get(0).getRootNode() === $el2.get(0).shadowRoot
```

Comment:
```suggestion
          $el1.get(0).getRootNode() === $el2?.get(0).shadowRoot
```

`$el2` could possibly be undefined, in which case this would throw an exception.

Reply:
Done